dowhy/causal_refuters/add_unobserved_common_cause.py

```diff
@@ -197,3 +198,191 @@ def include_confounders_effect(self, new_data, kappa_t, kappa_y):
         return new_data
+
+    def include_simulated_confounder(self, sample_size, given_data, convergence_threshold):
+ 
+        required_variables = True
+        observed_variables = self.choose_variables(required_variables)
+
+        new_data = given_data
+        observed_variables_with_treatment_and_outcome = observed_variables + self._treatment_name + self._outcome_name
+        new_data = new_data[observed_variables_with_treatment_and_outcome]
+        new_data[self._treatment_name[0]] = new_data[self._treatment_name[0]].astype('int64')
+
+        #outcome model 
+        y = new_data[self._outcome_name[0]]
+        l = observed_variables + self._treatment_name
+        X = new_data[l]
+        
+        model = sm.OLS(y,X)
+        results = model.fit()
+        standardized_residuals_y = y - results.fittedvalues
+
+        #treatment model 
+        t = new_data[self._treatment_name[0]].astype('int64')
+        l = observed_variables
+        X = new_data[l]
+        model = sm.OLS(t,X)
+        results = model.fit()
+        standardized_residuals_z = t - results.fittedvalues
+
+        max_correlation_with_y = -10000000000
+        max_correlation_with_z = 0
+
+        for i in observed_variables:
+            column1 = given_data[i]
+            column2 = given_data[self._outcome_name[0]]
+            correlation_y = column1.corr(column2)
+            if correlation_y>=max_correlation_with_y:
+                max_correlation_with_y = correlation_y
+                column3 = new_data[self._treatment_name[0]]
+                max_correlation_with_z = column1.corr(column3)
+
+        correlation_y_list = []
+        correlation_z_list = []
+        new_metric_simulated_list = []    
+        x_list = []
+
+        for i in range(0, 1000, 100):
+            c1 = math.sqrt(i)
+            c2 = c1
+            U = []
+            d_y = list(pd.Series(standardized_residuals_y))
+            d_z = list(pd.Series(standardized_residuals_z))
+            for j in range(len(d_z)):
+                simulated_variable_mean = c1*d_y[j]+c2*d_z[j]
+                simulated_variable_stddev = 0.01
+                U.append(np.random.normal(simulated_variable_mean, simulated_variable_stddev, 1))
+            U = np.array(U)
+            model = sm.OLS(U,X)
+            results = model.fit()
+            U = U.reshape(-1, )
+            final_U = U - results.fittedvalues.values
+            new_data['simulated'] = final_U
+            column1 = new_data['simulated']
+            column2 = given_data[self._outcome_name[0]]
+            correlation_y = column1.corr(column2)
+            correlation_y_list.append(correlation_y)
+
+            column3 = new_data[self._treatment_name[0]]
+            correlation_z = column1.corr(column3)
+            correlation_z_list.append(correlation_z)
+
+            new_metric_simulated = correlation_y*correlation_z
```

Comment:
Is there a better name than `new_metric`? Perhaps `product_cor_metric`. The more informative the variable names, the better.